Doc/library/pathlib.rst

```diff
@@ -1539,50 +1539,32 @@ Creating files and directories
 Copying, renaming and deleting
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-.. method:: Path.copy(target, *, follow_symlinks=True, preserve_metadata=False)
+.. method:: Path.copy(target, *, follow_symlinks=True, dirs_exist_ok=False, \
+                      preserve_metadata=False, ignore=None, on_error=None)
 
-   Copy the contents of this file to the *target* file. If *target* specifies
-   a file that already exists, it will be replaced.
+   Copy this file or directory tree to the given *target*.
 
-   If *follow_symlinks* is false, and this file is a symbolic link, *target*
-   will be created as a symbolic link. If *follow_symlinks* is true and this
-   file is a symbolic link, *target* will be a copy of the symlink target.
+   If the source is a file, the target will be replaced if it is an existing
+   file. If the source is a symlink and *follow_symlinks* is true (the
+   default), the symlink's target is copied. Otherwise, the symlink is
+   recreated at the destination.
 
-   If *preserve_metadata* is false (the default), only the file data is
-   guaranteed to be copied. Set *preserve_metadata* to true to ensure that the
-   file mode (permissions), flags, last access and modification times, and
-   extended attributes are copied where supported. This argument has no effect
-   on Windows, where metadata is always preserved when copying.
+   If the source is a directory and *dirs_exist_ok* is false (the default), a
+   :exc:`FileExistsError` is raised if the target is an existing directory.
+   If *dirs_exists_ok* is true, the copying operation will continue if it
+   encounters existing directories, and files within the destination tree will
```

Comment:
This sentence feels a bit longer than necessary. Maybe "Otherwise, the files within..." or "If *dirs_exists_ok* is true, the files within..."

Reply:
I think it's a tiny bit ambiguous if we leave out "the copying operating will continue _if it encounters existing directories_", because the bit I've italicised isn't part of the main condition in the first sentence, so it won't necessarily be assumed by the reader here.

Reply:
The current wording seems fine. To be honest I'm not sure I understand the point of the condition (*if it encounters existing directories*), because surely this follows from not failing if a directory is encountered. So you could have

```
   If *dirs_exists_ok* is true, the copying operation will overwrite
   existing files within the destination tree with corresponding files
   from the source tree.
```